x-pack/plugin/sql/qa/single-node/src/test/java/org/elasticsearch/xpack/sql/qa/single_node/JdbcCsvNestedDocsIT.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.qa.single_node;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.xpack.sql.qa.jdbc.CsvTestUtils.CsvTestCase;
+import org.elasticsearch.xpack.sql.qa.jdbc.SpecBaseIntegrationTestCase.Parser;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.xpack.sql.qa.jdbc.CsvTestUtils.specParser;
+
+/**
+ * Test class to check https://github.com/elastic/elasticsearch/issues/35176 fix
+ */
+public class JdbcCsvNestedDocsIT extends JdbcCsvSpecIT {
```

Comment:
No need to create a separate class - the logic here can be folded into `JdbcCsvSpecIT` which would override `executeJdbcQuery` and if the test is `nested`, use a smaller max random. To make the change smaller, add a maximumFetchSize() method in `SpecBaseIntegrationTest` which then gets used inside `executeJdbcQuery`.